lisa/tools/sysbench.py

```diff
@@ -0,0 +1,563 @@
+            r"events \(avg/stddev\):\s+(?P<events_avg>[\d.]+)/(?P<events_stddev>[\d.]+)"
+        )
+        thread_events = thread_events_pattern.search(non_debug_lines)
+        if thread_events:
+            result[thread_param]["events_avg"] = thread_events.group("events_avg")
+            result[thread_param]["events_stddev"] = thread_events.group("events_stddev")
+
+        thread_exec_time_pattern = re.compile(
+            r"execution time \(avg/stddev\):\s+(?P<avg>[\d.]+)/(?P<std_dev>[\d.]+)"
+        )
+        exec_time = thread_exec_time_pattern.search(non_debug_lines)
+        if exec_time:
+            result[thread_param]["execution_time_avg"] = exec_time.group("avg")
+            result[thread_param]["execution_time_stddev"] = exec_time.group("std_dev")
+
+        # Get Verbose per-thread statistics
+        verbose_param = "verbose_per_thread_statistics"
+        verbose_metrics_pattern = re.compile(
+            r"(?P<metric>min|avg|max|events):\s+(?P<value>[\d.]+)"
+        )
+        result[verbose_param] = {}
+
+        matches = verbose_metrics_pattern.findall(debug_lines)
+        for match in matches:
+            if match:
+                metric = match[0]
+                value = float(match[1])
+                result[verbose_param][metric] = value
+
+        event_execution_pattern = re.compile(
+            r"total time taken by event execution:\s+(?P<time>[\d.]+s)"
+        )
+        match = event_execution_pattern.search(debug_lines)
+        if match:
+            result[verbose_param]["event_execution_time"] = match.group("time")
+
+        return result
+
+    def __process_memory_perf_result(
+        self,
+        data: str,
+    ) -> Dict[Any, Any]:
+        result: Dict[Any, Any] = self.__process_perf_result(data)
+
+        total_operations = None
+        operations_per_second = None
+        total_mib_transferred = None
+        mib_per_second = None
+
+        # Extract Total operations and operations per second
+        # Sample Output
+        # Total operations: 65730837 (6571036.01 per second)
+        total_operations_pattern = re.compile(
+            r"Total operations: (?P<operation>\d+) \((?P<per_sec>[\d.]+) per second\)"
+        )
+        match = total_operations_pattern.search(data)
+        if match:
+            total_operations = int(match.group("operation"))
+            operations_per_second = float(match.group("per_sec"))
+
+        # Extract Total MiB transferred and MiB per second
+        # Sample Output
+        # 64190.27 MiB transferred (6417.03 MiB/sec)
+        total_mib_transferred_pattern = re.compile(
+            r"(?P<total_mib>[\d.]+) MiB transferred \((?P<per_sec>[\d.]+) MiB/sec\)"
+        )
+        match = total_mib_transferred_pattern.search(data)
+        if match:
+            total_mib_transferred = float(match.group("total_mib"))
+            mib_per_second = float(match.group("per_sec"))
+
+        result["total_operations"] = total_operations
```

Comment:
Follow other perf messages to create a mem perf message. The result dict is not easy to query or compare in dashboards.

Reply:
Done. Please check latest push